python/hsfs/engine/python.py

```diff
@@ -933,17 +925,20 @@ def acked(err, msg):
         if offline_write_options is not None and offline_write_options.get(
             "start_offline_backfill", True
         ):
-            print("Launching offline feature group backfill job...")
-            self._job_api.launch(job_name)
-            print(
-                "Backfill Job started successfully, you can follow the progress at \n{}".format(
-                    self._get_job_url(job.href)
-                )
-            )
-            self._wait_for_job(job, offline_write_options)
+            self.run_job(job, offline_write_options)
```

Comment:
If you do what I suggest in the other comments, you should be able to replace this with

```python
job.run(await_termination=write_options.get("wait_for_job", True))
```